Realizar un modelo lineal para el siguiente problema de predicción de demanda de energía 

| Velocidad del viento | Personas dentro del edificio | Requerimiento de energía |
|-----------------------|------------------------------|---------------------------|
| 100                   | 2                            | 5                         |
| 50                    | 42                           | 25                        |
| 45                    | 31                           | 22                        |
| 60                    | 35                           | 18                        |


In [2]:
import numpy as np
viento = np.array([100, 50, 45, 60])
personas = np.array([2, 42, 31, 35])
energia = np.array([5, 25, 22, 18])

In [3]:
X = np.column_stack((np.ones(len(viento)), viento, personas))
y = energia.reshape(-1, 1)

# Fórmula de mínimos cuadrados: θ = (X^T X)^(-1) X^T y
theta = np.linalg.inv(X.T @ X) @ X.T @ y

print("Parámetros del modelo:")
print(theta)

Parámetros del modelo:
[[24.97615881]
 [-0.20751687]
 [ 0.20920152]]


Por lo que el modelo queda de la siguiente manera :

$ \hat{y}= \theta_{0} + \theta_{1} x_{1} + \theta_{2} x_{2}$

$ \hat{y}=$ 24.98−0.207⋅(viento)+0.209⋅(personas)